tests/FakeItEasy.Specs/SerializationSpecs.cs

```diff
@@ -0,0 +1,136 @@
+namespace FakeItEasy.Specs
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.IO;
+    using System.Runtime.Serialization;
+    using System.Runtime.Serialization.Formatters.Binary;
+    using FakeItEasy.Sdk;
+    using FakeItEasy.Tests.TestHelpers;
+    using FluentAssertions;
+    using Xbehave;
+    using Xunit;
+
+    public static class SerializationSpecs
+    {
+        [SuppressMessage("Microsoft.Design", "CA1040:AvoidEmptyInterfaces", Justification = "It's just used for testing.")]
+        public interface IInterface
+        {
+            int ParseAString(string someString);
+        }
+
+        [Scenario]
+        [InlineData(typeof(IInterface))]
+        [InlineData(typeof(AbstractClass))]
+        [InlineData(typeof(ClassWithProtectedConstructor))]
+        [InlineData(typeof(ClassWithInternalConstructorVisibleToDynamicProxy))]
+        [InlineData(typeof(InternalClassVisibleToDynamicProxy))]
+        public static void SerializingSupportedTypes(Type typeOfFake, object fake)
+        {
+            "Given a serializable fake type"
+                .See(typeOfFake.ToString());
+
+            "When I create a fake of the supported type"
+                .x(() => fake = Create.Fake(typeOfFake));
+
+            "Then the fake can be serialized and deserialized"
+                .x(() => SerializeAndDeserialize(fake).Should().BeAssignableTo(typeOfFake));
+        }
+
+        // A characterization test, representing the current capabilities of the code, not the desired state.
+        [Scenario]
+        public static void SerializingWithCompletedCalls(IInterface fake, IInterface deserializedFake)
+        {
+            "Given a serializable fake"
+                .x(() => fake = A.Fake<IInterface>());
+
+            "And I call a method"
+                .x(() => fake.ParseAString("a string"));
+
+            "When I serialize and deserialize the fake"
+                .x(() => deserializedFake = SerializeAndDeserialize(fake));
+
+            "Then the record of the completed call is lost"
+                .x(() => Fake.GetCalls(deserializedFake).Should().BeEmpty());

```

Comment:
@thomaslevesque, this is the test that I had to update. Now that I don't serialize the recorded calls on the FakeManager, serialization passes. Previously it would've thrown on serialization.
However, we still don't have access to the previously-recorded calls upon deserialization.